pywps/app/Process.py

```diff
@@ -267,15 +267,20 @@ def launch_next_process(self):
     def clean(self):
         """Clean the process working dir and other temporary files
         """
-        LOGGER.info("Removing temporary working directory: %s" % self.workdir)
-        try:
-            if os.path.isdir(self.workdir):
-                shutil.rmtree(self.workdir)
-            if self._grass_mapset and os.path.isdir(self._grass_mapset):
-                LOGGER.info("Removing temporary GRASS GIS mapset: %s" % self._grass_mapset)
-                shutil.rmtree(self._grass_mapset)
-        except Exception as err:
-            LOGGER.error('Unable to remove directory: %s', err)
+        testmode = config.get_config_value('server', 'testmode')
+        if testmode is True:
+            LOGGER.warning('PyWPS is running in testing mode. Workdir is not removed: {}'.format(
+                self.workdir))
+        else:
+            LOGGER.info("Removing temporary working directory: %s" % self.workdir)
+            try:
+                if os.path.isdir(self.workdir):
+                    shutil.rmtree(self.workdir)
+                if self._grass_mapset and os.path.isdir(self._grass_mapset):
+                    LOGGER.info("Removing temporary GRASS GIS mapset: %s" % self._grass_mapset)
+                    shutil.rmtree(self._grass_mapset)
+            except Exception as err:
+                LOGGER.error('Unable to remove directory: %s', err)
```

Comment:
What's the canonical way of formatting strings in `logging` statements espeically in a Python 3 context?  Is `%s` ok?  Would be valuable to know for other projects as well.

Reply:
Can't say what the official recommendation is. Change to traditional style and updated log message.

Reply:
If the `{}` / `.format()` approach works, can we go with that to be more future proof?

Reply:
I think this is a topic for a next PR. But I haven't found an official statement concerning the logging. But I like the `format()` approach ;)

Reply:
I have opened an issue for this question #427.